code/modules/organs/organ_external.dm

```diff
@@ -567,202 +574,103 @@ Note that amputating the affected organ does in fact remove the infection from t
-			if(2)
-				owner.visible_message(
-					"<span class='danger'>\The [owner]'s [display_name] explodes in a shower of gore!</span>",\
-					"<span class='moderate'><b>Your [display_name] explodes in a shower of gore!</b></span>",\
-					"<span class='danger'>You hear the sickening splatter of gore.</span>")
-				var/obj/effect/decal/cleanable/blood/gibs/gore = new(get_turf(owner))
-				if(owner.species.flesh_color)
-					gore.fleshcolor = owner.species.flesh_color
-				if(owner.species.blood_color)
-					gore.basecolor = owner.species.blood_color
-				gore.update_icon()
-				gore.throw_at(get_edge_target_turf(owner,pick(alldirs)),rand(1,3),30)
-
-				if(body_part == HEAD && owner.brain_op_stage != 4.0)
-					var/mob/living/simple_animal/borer/borer = owner.has_brain_worms()
-					if(borer)
-						borer.detatch() //Should remove borer if the brain is removed - RR
-					if(owner.species && owner.species.flags & IS_SYNTHETIC)
-						var/obj/item/device/mmi/posibrain/P = new(owner.loc)
-						P.transfer_identity(owner)
-						if(istype(P.loc,/turf))
-							P.throw_at(get_edge_target_turf(owner,pick(alldirs)),rand(1,3),30)
-					else
-						var/obj/item/brain/B = new(owner.loc)
-						B.transfer_identity(owner)
-						if(istype(B.loc,/turf))
-							B.throw_at(get_edge_target_turf(owner,pick(alldirs)),rand(1,3),30)
-
-
-
-		if(organ)
-			if(!amputation)
-				owner.visible_message(
-					"<span class='danger'>\The [owner]'s [display_name] flies off in an arc!</span>",\
-					"<span class='moderate'><b>Your [display_name] goes flying off!</b></span>",\
+					"<span class='danger'>\The [owner]'s [src.name] flies off in an arc!</span>",\
+					"<span class='moderate'><b>Your [src.name] goes flying off!</b></span>",\
 					"<span class='danger'>You hear a terrible sound of ripping tendons and flesh.</span>")
-			//Throw organs around
-			var/lol = pick(cardinal)
-			step(organ,lol)
-			// OK so maybe your limb just flew off, but if it was attached to a pair of cuffs then hooray! Freedom!
-			release_restraints()
-
-		if(vital)
-			owner.death()
+		if(DROPLIMB_BURN)
+			owner.visible_message(
+				"<span class='danger'>\The [owner]'s [src.name] flashes away into ashes!</span>",\
+				"<span class='moderate'><b>Your [src.name] flashes away into ashes!</b></span>",\
+				"<span class='danger'>You hear the crackling sound of burning flesh.</span>")
+		if(DROPLIMB_BLUNT)
+			owner.visible_message(
+				"<span class='danger'>\The [owner]'s [src.name] explodes in a shower of gore!</span>",\
+				"<span class='moderate'><b>Your [src.name] explodes in a shower of gore!</b></span>",\
+				"<span class='danger'>You hear the sickening splatter of gore.</span>")
+
+	src.removed(null, ignore_children)
+	owner.traumatic_shock += 60
+
+	wounds.Cut()
+	if(parent)
+		var/datum/wound/W
+		if(clean || max_damage < 50)
+			W = new/datum/wound/lost_limb/small(max_damage)
+		else
+			W = new/datum/wound/lost_limb(max_damage/2)
```

Comment:
Yikes that feels pretty gamey. The question is, how much health would the player potentially heal for, against what kind of downside? If it takes him 20 hits to chop off his own limb, those are 20 hits he can't give to someone else....I think it's worth leaving it as is and testing, and removing it if it's a problem.